### PR TITLE
Fixes #2996 - Improved Blog page

### LINKED ISSUE #2996: Improvable blog page.
**Actual Behaviour**

The current blog found at [https://susi.ai/blog](url) is messy and contains a heap of blog posts in a single page. If a user wants to access an older blog he/she has to scroll through all the recent blog posts before making it to the required part as their is no efficient way access the required blog post. The buttons that are used to share the posts at the bottom are not appealing either as they don't look and feel like buttons.

**Expected Behaviour**

The blog should be restructured in a way that the set of blog posts are displayed in a concise manner(without taking much space). All the blog posts should also be equally accessible without having to scroll through the entire page.

**Would you like to work on the issue?**

Yes.

### PR DESCRIPTION
**Pull request by AuraOfDivinity**
Fixes #2996 

Changes:
1) Added in a search bar to search for particular blog posts.
2) Blog posts initially have a small part of the post which could be expanded to view the entire post using the Read More button.
3) Changed the size and styling of the Facebook / Twitter sharing buttons to make them look more prominent.
4) Used the Raleway font within the blogposts.

Demo Link: https://pr-3023-fossasia-susi-web-chat.surge.sh

Screenshots of the change: 
ImprovedBlog

### CODE REVIEW

**Comment by AuraOfDivinity** [comment]
@akshatnitd @plxity The requested changes were made. Could you kindly have a look and let me know if anything else should be done?

File: src/components/About/Blog/index.js [hunk 1]
```diff
@@ -381,24 +439,92 @@ class Blog extends Component {
...
+                        <Overlay>
+                          <OverlayLink href={posts.link}>
+                            {`Published on ${dateFormat(
+                              d,
+                              'dddd, mmmm dS, yyyy',
+                            )}`}
+                          </OverlayLink>
+                        </Overlay>
+                        {this.state.expandedElements.includes(i) ? (
+                          <Fade in={true} timeout={1100}>
```
Comment (akshatnitd) [inline src/components/About/Blog/index.js]: Please refactor out each Blog card into a separate file and have the expanded logic inside that file.

File: src/components/About/Blog/index.js [hunk 2]
```diff
@@ -280,6 +281,41 @@ class Blog extends Component {
...
+    this.expandedElements.splice(this.expandedElements.indexOf(index), 1);
+    this.setState({
+      expandedElements: this.expandedElements,
+    });
+  };
+
+  handleSearch = value => {
+    let displayCount = 0;
+    const { actions } = this.props;
+    var list = document.getElementsByClassName('section_blog');
```
Comment (akshatnitd) [inline src/components/About/Blog/index.js]: We should not use these APIs like `getElementsByClassName`. Please refrain from using these. You can directly use this.state.posts

**Comment by AuraOfDivinity** [comment]
@akshatnitd Closing because of https://github.com/fossasia/susi.ai/pull/3076

**Pull request closed without merging**